### DISCUSSION TITLE
Interesting Specimens

### DISCUSSION THREAD

**Discussion by katestange** [numberscope/frontscope#477] [General]
For specimens that are not necessarily for the gallery, or for anything in particular, right now, but that are fun and interesting and we shouldn't forget them.

Example, mousing-over this one is fun: 
http://localhost:5173/?name=FactorFence&viz=FactorFence&seq=OEIS+A001764

**Comment by katestange** [comment]
The beatty sequence for golden ratio shows the slowness with which Virahanka-Fibonacci moduli fill out:
http://localhost:5173/?name=Formula&viz=ModFill&modDimension=200&seq=Formula&formula=floor%28n*%281%2Bsqrt%285%29%29%2F2%29

**Reply by katestange** [reply]
And here's a very simple modfill that is lovely to watch the animation of:
http://localhost:5173/?name=Formula&viz=ModFill&modDimension=700&seq=Formula&formula=311*n

**Comment by gwhitney** [comment]
Here's a Turtle visualization of the sum of the terms of the [eta-expansion](https://en.wikipedia.org/wiki/Riemann_zeta_function#Equivalencies) of the Riemann zeta function, along the critical line as the imaginary part increases. I have spot-checked numerically, and it really is computing the zeta function in that the star is bouncing all around the correct value, up to a reflection about the real line which may either stem from p5.js coordinates being reversed vertically or in my borking a sign somewhere, not certain but I don't think it's too crucial. 

It's not so pretty though because this alternating-series version converges well on the critical line in the sense of quickly starting to oscillate around the true value (it doesn't actually converge quickly in the sense of finite partial sums with few terms being almost equal to the true value -- the oscillations go on and on at sqrt(n) magnitude, which is not conducive to "landing", so to speak).

http://localhost:5173/?name=Critical+Line+Zeta+via+Eta&viz=Turtle&speed=10&ruleMode=1&turnFormula=%28n+%3F+log1p%281%2Fn%29%281%2B0.01f%29+%3A+arg%281-2%5E%281%2F2-%281%2B0.01f%29i%29%29%29+*+360%2Ftau&stepFormula=200%28-1%29%5En%2F%28sqrt%28n%2B1%29*abs%281-2%5E%281%2F2-%281%2B0.01f%29i%29%29%29&colorFormula=rainbow%282b%29&seq=Formula&last=99&length=100

**Comment by gwhitney** [comment]
From Kate: We should be able to create a turtle preset that shows the space filling curve here: https://open.substack.com/pub/apieceofthepi/p/constructing-space-filling-curves?r=semnj&utm_campaign=post&utm_medium=email (see the colourful one at the bottom)

**Reply by gwhitney** [reply]
I agree. It would be sort of nice if the turtle traced out the first iteration, and then drew the second iteration over it in a different color, and so on. (Maybe that's what the colorful one at the bottom looks like; I can't see it because I am not a substack subscriber.)

**Comment by gwhitney** [comment]
From Kate: Also fun: http://localhost:5173/?name=Countdown&viz=Turtle&bgColor=22274e&speed=10&ruleMode=1&turnFormula=30%2B15*a&stepFormula=50&widthFormula=2&colorFormula=chroma%28%27skyblue%27%29.set%28%27hsl.h%27%2C+n%25360%29&seq=Formula&formula=%28n%5E2-n%2B2%29%2543

**Comment by gwhitney** [comment]
I posted this one on illustrating math because I am not really sure why it closes; replace 729 with 727 (or most other integers) and of course you get a periodic pattern but it rarely seems to close, whereas it closes with all the powers of 3 that I tried...:

http://localhost:5173/?name=Powers+of+Two+mod+a+Power+of+Three&viz=Turtle&angleMeaning=1&bgColor=22274e&ruleMode=1&angleFormula=%28n+%3F+2b+%3A+fraction%28360%2C729%29%29+%25+360&colorFormula=rainbow%28b%29&seq=Formula&last=729&length=730

**Reply by gwhitney** [reply]
P.S.  Oh I did actually figure out why it closes. The pattern has a multiplier a (here 2) and a denominator d (here 729=3^6). If a is a primitive root mod p, then if the denominator is p the turtle's path sums all primitive pth roots of unity, which sum to -1, so the path does not close; whereas if the denominator in p^k the path sums all primitive (p^k)th roots of unity, which sum to the negative of _all_ (p^(k-1))th roots of unity, which sum to 0, so the path closes.

**Comment by katestange** [comment]
Some thoughts:

1. Folding rates, under bearing setting, has a bit different effect; maybe the description should be updated slightly to indicate it is changing the angle or bearing, as appropriate.

2. Here's the Gosper Flowsnake:  http://localhost:5173/?name=The+Gosper+Flowsnake&viz=Turtle&domain=0+1+2+3+4+5+6&angles=0+0+60+120+-60+-120+180&angleMeaning=1&steps=8&widths=2&strokeColor=%2390EE90+%2300CED1+%23FF7D4D+%231F93FF+%23B63A71+%23C71585+%23FF7D4D&bgColor=22274e&speed=10&angleFormula=%7B%220%22%3A+0%2C+%221%22%3A+0%2C+%222%22%3A+60%2C+%223%22%3A+120%2C+%224%22%3A+-60%2C+%225%22%3A+-120%2C+%226%22%3A+180%7D%5Bstring%28a%29%5D&stepFormula=%7B%220%22%3A+8%2C+%221%22%3A+8%2C+%222%22%3A+8%2C+%223%22%3A+8%2C+%224%22%3A+8%2C+%225%22%3A+8%2C+%226%22%3A+8%7D%5Bstring%28a%29%5D&widthFormula=%7B%220%22%3A+2%2C+%221%22%3A+2%2C+%222%22%3A+2%2C+%223%22%3A+2%2C+%224%22%3A+2%2C+%225%22%3A+2%2C+%226%22%3A+2%7D%5Bstring%28a%29%5D&colorFormula=%7B%220%22%3A+%2390EE90%2C+%221%22%3A+%2300CED1%2C+%222%22%3A+%23FF7D4D%2C+%223%22%3A+%231F93FF%2C+%224%22%3A+%23B63A71%2C+%225%22%3A+%23C71585%2C+%226%22%3A+%23FF7D4D%7D%5Bstring%28a%29%5D&seq=OEIS+A229214&modulus=7&last=1040&length=1040  ; I wish it could be more centered.

3. Another space-filling curve I came across in OEIS: http://localhost:5173/?name=OEIS+A000005&viz=Turtle&domain=0+1+2+3&angles=0+0+120+-120&angleMeaning=1&steps=15&widths=2&strokeColor=%2390EE90+%2300CED1+%23FF7D4D+%231F93FF+%23B63A71+%23C71585+%23FF7D4D&bgColor=22274e&speed=10&angleFormula=%7B%220%22%3A+0%2C+%221%22%3A+0%2C+%222%22%3A+120%2C+%223%22%3A+-120%7D%5Bstring%28a%29%5D&stepFormula=%7B%220%22%3A+15%2C+%221%22%3A+15%2C+%222%22%3A+15%2C+%223%22%3A+15%7D%5Bstring%28a%29%5D&widthFormula=%7B%220%22%3A+2%2C+%221%22%3A+2%2C+%222%22%3A+2%2C+%223%22%3A+2%7D%5Bstring%28a%29%5D&colorFormula=%7B%220%22%3A+%2390EE90%2C+%221%22%3A+%2300CED1%2C+%222%22%3A+%23FF7D4D%2C+%223%22%3A+%231F93FF%7D%5Bstring%28a%29%5D&seq=OEIS+A265671

4. The Angle field is a little flaky.  Delete out everything and it populates NaN and you have to delete those three characters to get an empty field, but then that stays empty.  More annoying and odd, if you type 0 30 -90 10 and then arrow key over to sit immediately after the -90 and backspace twice, the cursor pops to the end *and* the - disappears.  You might have wanted to backspace the 90 to put an 80... it won't behave the way you expect.  This behaviour is, however, intermittent!  It did this for me (repeatedly) in chrome and then when I tabbed over here to firefox and back, it stopped doing it quite like that -- now it would only pop to the end when deleting the - also.  I'm sorry, it's weird.  But I think we had decided that fields should not "jump about"/"self-correct" under the user's typing as a general rule, so maybe we need to just turn something back off.

**Reply by gwhitney** [reply]
> 2. Here's the Gosper Flowsnake; I wish it could be more centered.

Here's a slightly more economical URL: http://localhost:5173/?name=The+Gosper+Flowsnake&viz=Turtle&domain=-3+-2+-1+1+2+3&angles=-60+-120+180+0+60+120&angleMeaning=1&steps=8&widths=2&strokeColor=%2390EE90+%2300CED1+%23FF7D4D+%231F93FF+%23B63A71+%23C71585&bgColor=22274e&speed=10&angleFormula=%7B%22-3%22%3A+-60%2C+%22-2%22%3A+-120%2C+%22-1%22%3A+180%2C+%221%22%3A+0%2C+%222%22%3A+60%2C+%223%22%3A+120%7D%5Bstring%28a%29%5D&stepFormula=%7B%22-3%22%3A+8%2C+%22-2%22%3A+8%2C+%22-1%22%3A+8%2C+%221%22%3A+8%2C+%222%22%3A+8%2C+%223%22%3A+8%7D%5Bstring%28a%29%5D&widthFormula=%7B%22-3%22%3A+2%2C+%22-2%22%3A+2%2C+%22-1%22%3A+2%2C+%221%22%3A+2%2C+%222%22%3A+2%2C+%223%22%3A+2%7D%5Bstring%28a%29%5D&colorFormula=%7B%22-3%22%3A+%2390EE90%2C+%22-2%22%3A+%2300CED1%2C+%22-1%22%3A+%23FF7D4D%2C+%221%22%3A+%231F93FF%2C+%222%22%3A+%23B63A71%2C+%223%22%3A+%23C71585%7D%5Bstring%28a%29%5D&seq=OEIS+A229214&last=1040&length=1040

I think with pan and rotate the initial positioning isn't too crucial , and there's nothing we can do about the order that this curve visits points. But if you feel we should implement an optional start position for Turtle, please do file an issue for it.

**Reply by gwhitney** [reply]
> 3. Another space-filling curve I came across in OEIS

Ooh, this one looks sort of like a triangular-grid dragon curve.

**Reply by gwhitney** [reply]
And here is the dragon curve itself: http://localhost:5173/?name=OEIS+A014577&viz=Turtle&domain=0+1&angles=-90+90&steps=5&speed=8&angleFormula=%7B%220%22%3A+-90%2C+%221%22%3A+90%7D%5Bstring%28a%29%5D&stepFormula=%7B%220%22%3A+5%2C+%221%22%3A+5%7D%5Bstring%28a%29%5D&widthFormula=%7B%220%22%3A+1%2C+%221%22%3A+1%7D%5Bstring%28a%29%5D&colorFormula=%7B%220%22%3A+%23c98787%2C+%221%22%3A+%23c98787%7D%5Bstring%28a%29%5D&seq=OEIS+A014577

**Comment by katestange** [comment]
5.  If you get NaN in turn angles and don't notice, but then go to "Formula" mode (maybe that's all you wanted anyway), you then have an error (from the NaN) sitting up top of the visualizer panel and nothing works right (colouring doesn't work, canvas turns back to default colour), but the offending field is hidden.  Not sure what to suggest here, but it did cause me confusion for a bit (particularly the canvas turning colour).  Should the error checking on hidden "List" variables be turned off when they are hidden?

**Comment by katestange** [comment]
Here's the Hat Monotile: http://localhost:5173/?name=Hat+Monotile&viz=Turtle&domain=0+1+2+3&angles=0&steps=15&widths=2&strokeColor=%2390EE90+%2300CED1+%23FF7D4D+%231F93FF+%23B63A71+%23C71585+%23FF7D4D&bgColor=795767&speed=10&ruleMode=1&angleFormula=a*30&widthFormula=3&colorFormula=chroma.scale%28%27YlGnBu%27%29.colors%2850%29%5Bmod%28n%2C30%29%2B1%5D&colorChooser=409CF5&seq=OEIS+A363348

**Comment by katestange** [comment]
A possible contender for gallery:  http://localhost:5173/?name=The+Raucous+Recaman&viz=Turtle&domain=0+1&steps=2+3&strokeColor=%235e8d85&bgColor=E5E0F6&speed=20&ruleMode=1&angleFormula=89.5&stepFormula=a%2F30&colorFormula=rainbow%28n%29&colorChooser=813d9c&seq=OEIS+A005132

**Comment by katestange** [comment]
This doesn't produce anything... but it fails silently:  http://localhost:5173/?name=OEIS+A000521&viz=Turtle&domain=0+1&steps=2+3&strokeColor=%235e8d85&bgColor=000000&speed=100&ruleMode=1&angleFormula=a+%25+30&stepFormula=3&widthFormula=6&colorFormula=chroma%28%7Bh%3A+%28n%2F1.5%29+%25+360%2C+s%3A0.65%2C+l%3A+0.75%7D%29&seq=OEIS+A000521
By contrast, this produces something... http://localhost:5173/?name=OEIS+A000521&viz=Turtle&domain=0+1&steps=2+3&strokeColor=%235e8d85&bgColor=000000&speed=100&ruleMode=1&angleFormula=a+%25+30+%3D%3D+0+%3A+30+%3A+0&stepFormula=3&widthFormula=6&colorFormula=chroma%28%7Bh%3A+%28n%2F1.5%29+%25+360%2C+s%3A0.65%2C+l%3A+0.75%7D%29&seq=OEIS+A000521
The only difference is in the angle formula.  Bigint issue?  I think once the coefficients become bigints, the comparison always fails (we get a straight line).  The picture doesn't seem right.

**Comment by katestange** [comment]
Mobius is fun to play with http://localhost:5173/?name=Mobius+wandering&viz=Turtle&domain=0+1&steps=2+3&strokeColor=%235e8d85&bgColor=B1AEEA&speed=40&ruleMode=1&angleFormula=n*90&stepFormula=a*5&widthFormula=4&colorFormula=chroma%28%7Bh%3A+%28n%2F1.50%29+%25+360%2C+s%3A0.65%2C+l%3A+0.75%7D%29&seq=OEIS+A008683

**Reply by gwhitney** [reply]
Oh, interesting that you didn't just turn by k*a and always step 5, as opposed to always turn k and step by a*l. Both wander, albeit differently looking.. Oh wait, you're not turning by a constant amount, you're turning left, about face, turning right, not turning in a cycle. Hmmm. You also get a wandering by always turning k. Any theoretical reason to prefer one of these wanderings over another, or is it purely the aesthetics of the results?

**Comment by katestange** [comment]
I believe this is a Gauss sum (using info from https://oeis.org/A307380 although the sequence pane doesn't matter!)
http://localhost:5173/?name=Gauss+sum&viz=Turtle&domain=0+1&angleMeaning=1&steps=2+3&strokeColor=%235e8d85&bgColor=070707&speed=20&ruleMode=1&angleFormula=2*pi*%28n%2F11+%2B+%5B0%2C+0%2C+1%2C+3%2C+2%2C+4%2C+4%2C+2%2C+3%2C+1%2C+0%5D%5Bn%2511%2B1%5D%2F5%29&angleMeasure=1&stepFormula=15&colorFormula=chroma%28%7Bh%3A+%28n%2F1.50%29+%25+360%2C+s%3A0.65%2C+l%3A+0.75%7D%29&seq=Formula

**Reply by gwhitney** [reply]
This just zigzags off in a uniform direction periodically forever, or so it appears... is that what you expected?

**Comment by katestange** [comment]
Gaussian periods: http://localhost:5173/?name=Gaussian+periods&viz=Turtle&domain=0+1&angleMeaning=1&steps=2+3&strokeColor=%235e8d85&bgColor=070707&speed=100&ruleMode=1&angleFormula=2*pi*n%5E2%2F2971&angleMeasure=1&stepFormula=1&colorFormula=chroma%28%7Bh%3A+%28n%2F1.50%29+%25+360%2C+s%3A0.65%2C+l%3A+0.75%7D%29&seq=Formula

**Comment by gwhitney** [comment]
Here's the trinomial equivalent of the usual arithmetic triangle coloring by residue; this one is mod 3. It's a nice example because it naturally uses the triangle shape (we didn't have any other example of that) and it uses a custom fill order. Also, it's familiar yet a bit different.

http://localhost:5173/?name=Triscal%27s+Triangle&viz=FormulaGrid&dimensions=25&aspect=r&fillOrder=3&pathFormula=%5B2*ceil%28c%2F4%29+-+floor%28sqrt%28k-1%29%29+%2B+k+-+floor%28sqrt%28k-1%29%29%5E2%2C+floor%28sqrt%28k-1%29%29%2B+1%5D&backgroundColor=21218FFF&fillFormula=%7Btriangle%3A+rainbow%28120a%29%2C+text%3A+a%7D&seq=OEIS+A027907

**Comment by gwhitney** [comment]
Here's an attractive Ulam spiral devised by @Vectornaut with a "background" that highlights the spiral path:
http://localhost:5173/?name=OEIS+A010051&viz=FormulaGrid&dimensions=70+70&aspect=1&fillOrder=1&pathFormula=spiral%28k%29&backgroundColor=000000FF&fillFormula=%7Brectangle%3A+chroma.lch%2832+%2B+7*sin%285%2F9*pi*sqrt%28k%29%29%2C+31%2C+100*sqrt%28k%29%29%2C+circle%3A+chroma.lch%2875%2C+81%2C+72%29.alpha%28a%29%7D&inset=0.8&seq=OEIS+A010051

**Comment by gwhitney** [comment]
An interesting cubical-looking texture resulting from the overlapping hexagons:
http://localhost:5173/?name=Formula%3A+n&viz=FormulaGrid&dimensions=50&aspect=r&backgroundColor=21218FFF&fillFormula=%7Bhexagon%3A+rainbow%2810*sqrt%28%28x-c%2F2%29%5E2+%2B+%28y-r%2F2%29%5E2%29%2C+%28x%2By%29+%25+2+*+0.5+%2B+0.25%29%7D&seq=Formula

**Comment by gwhitney** [comment]
Here's a pretty variation on the usual chaos game:
http://localhost:5173/?name=Chaos+Game&viz=Chaos&corners=3&stepFormula=0.5++-+0.0018y&showLabels=true&seq=Random&max=2

**Reply by katestange** [reply]
Very nice!

**Comment by gwhitney** [comment]
Here's a list of OEIS sequences with "interesting plots" (I think more traditional plots, like scatterplots, "pin plots", log-log plots, those sorts of things) , I believe from a talk by Neil Sloane that I attended: A002487, A285687, A265326, A357082 (the "scariest sequence"), A117966, A063543, A229037, A055748, A137655
